inception-pdf-editor/src/main/java/de/tudarmstadt/ukp/inception/pdfeditor/pdfanno/PdfAnnoRenderer.java

```diff
@@ -108,35 +110,62 @@ public static PdfAnnoModel render(AnnotatorState aState, VDocument aVDoc, String
     private static Span convertToPdfAnnoSpan(VSpan aVspan, String aColor, String aDocumentText,
                                              PdfExtractFile aPdfExtractFile)
     {
-        VRange range = aVspan.getRanges().get(0); // TODO: handle multiple ranges
-        // use offset pre and post string to increase uniqueness of annotation text
-        int offset = 20;
-        int start = range.getBegin() <= offset ? 0 : range.getBegin() - offset;
-        int end = range.getEnd() < aDocumentText.length() - offset
-            ? range.getEnd() + offset : aDocumentText.length() - 1;
-
-        String annotatedText = aDocumentText.substring(start, end);
-        String preText = annotatedText.substring(0,
-            (range.getBegin() <= offset ? range.getBegin() - start : offset));
-        String postText = annotatedText.substring(annotatedText.length() -
-            (range.getEnd() < aDocumentText.length() - offset
-                ? offset : aDocumentText.length() - range.getEnd()), annotatedText.length()) ;
-        // remove whitespaces as they are not present in PDFExtract text
-        String cleanAnnotatedText = annotatedText.replaceAll("\\s", "");
-
-        int index = aPdfExtractFile.getStringContent().indexOf(cleanAnnotatedText);
-        if (index < 0) {
-            LOG.error("Could not map exisiting annotation with id " + aVspan.getVid().toString());
-            return null;
-        } else {
-            start = index + preText.replaceAll("\\s", "").length() + 1;
-            end = index + cleanAnnotatedText.length() - postText.replaceAll("\\s", "").length();
-            PdfExtractLine first = aPdfExtractFile.getStringPdfExtractLine(start);
-            PdfExtractLine last = aPdfExtractFile.getStringPdfExtractLine(end);
-            return new Span(aVspan.getVid().getId(), first.getPage(), aColor,
-                cleanAnnotatedText.substring(start - index, end - index).replaceAll("\\s", ""),
-                first.getPosition(), last.getPosition());
+        int docTextLen = aDocumentText.length();
+        int vSpanBegin = aDocumentText.length();
+        int vSpanEnd = -1;
+        // search for begin of the first range and end of the last range and use those
+        for (VRange range : aVspan.getRanges())
+        {
+            if (range.getBegin() < vSpanBegin) vSpanBegin = range.getBegin();
```

Comment:
What is the purpose of these if clauses?

Reply:
Finding the lowest begin and the highest end of the ranges.
Multiple ranges for one annotation in INCEpTION only occur when annotation crosses sentence boundary, right?
So what we do here is basically search for the begin and end of all ranges together so we can create one annotation for it.

Reply:
Whops - I thought I had cancelled this comment instead of submitting it. It took me a moment to understand what the code does and after that I also understood the line comment above the for loop. Maybe you would like to "simplify" the code a bit to something like (no idea if you think this is simpler):

```
        int begin = span.getRanges().stream()
            .mapToInt(range -> Math.min(range.getBegin(), range.getEnd()))
            .min().getAsInt();

```

Reply:
Sure, seems good :)